docs/source/dev-guide/plugins/environment_exporters.rst

```diff
@@ -0,0 +1,411 @@
+                    "md5": pkg.md5,
+                }
+                for pkg in environment.explicit_packages
+            ]
+
+        # Add environment variables
+        if environment.variables:
+            data["environment"]["variables"] = dict(environment.variables)
+
+        return json.dumps(data, indent=2, sort_keys=True)
+
+
+    @conda.plugins.hookimpl
+    def conda_environment_exporters():
+        yield conda.plugins.CondaEnvironmentExporter(
+            name="custom-json",
+            aliases=("cjson",),
+            default_filenames=("environment.cjson", "env.cjson"),
+            export=export_custom_json,
+        )
+
+Error Handling
+--------------
+
+Your export function should handle error cases appropriately:
+
+.. code-block:: python
+
+    from conda.exceptions import CondaValueError
+
+
+    def export_strict_format(environment: Environment) -> str:
+        """Export that requires specific conditions."""
+        if not environment.dependencies:
+            raise CondaValueError(
+                "Cannot export strict format: no dependencies found. "
+                "This format requires at least one dependency."
+            )
+
+        if not environment.name:
+            raise CondaValueError(
+                "Cannot export strict format: environment name is required."
+            )
+
+        # Continue with export...
+        return formatted_content
+
+Working with Different Package Types
+=====================================
+
+Understanding Package Collections
+---------------------------------
+
+The Environment model provides different package collections for different use cases:
+
+``dependencies`` (MatchSpec objects)
+  Represents user-requested packages. These are the packages the user explicitly
+  asked for, either from history (when using ``--from-history``) or converted
+  from installed packages.
+
+``explicit_packages`` (PackageRecord objects)
+  Represents all installed packages with full metadata including URLs, checksums,
+  and build information. Used for exact reproduction.
+
+Example usage patterns:
+
+.. code-block:: python
+
+    def export_user_requested(environment: Environment) -> str:
+        """Export only what the user explicitly requested."""
+        if not environment.dependencies:
+            raise CondaValueError("No requested packages found")
+
+        lines = []
+        for dep in environment.dependencies:
+            lines.append(str(dep))  # e.g., "numpy=1.21.0"
+        return "\n".join(lines)
+
+
+    def export_exact_reproduction(environment: Environment) -> str:
+        """Export for exact environment reproduction."""
+        if not environment.explicit_packages:
+            raise CondaValueError("No installed packages found")
+
+        lines = ["@EXPLICIT"]
+        for pkg in environment.explicit_packages:
+            lines.append(pkg.url)  # Full package URL
+        return "\n".join(lines)
+
+Plugin Registration and Distribution
```

Comment:
I think this is a common pattern for all the types of plugins. Should this section be at a higher up level doc, like a new doc docs/source/dev-guide/plugins/distribution?